algorithms/unit_tests/TestStdAlgorithmsTeamAdjacentDifference.cpp

```diff
@@ -0,0 +1,206 @@
+
+namespace Test {
+namespace stdalgos {
+namespace TestAdjacentDifference {
+
+namespace KE = Kokkos::Experimental;
+
+template <class ValueType>
+struct PlusFunctor {
+  KOKKOS_INLINE_FUNCTION constexpr ValueType operator()(
+      const ValueType& lhs, const ValueType& rhs) const {
+    return lhs + rhs;
+  }
+};
+
+template <class ViewType, class DiffsViewType, class BinaryOp>
+struct TestFunctorA {
+  ViewType m_view;
+  DiffsViewType m_diffsView;
+  int m_apiPick;
+  BinaryOp m_binaryOp;
+
+  TestFunctorA(const ViewType view, const DiffsViewType diffsView, int apiPick,
+               BinaryOp binaryOp)
+      : m_view(view),
+        m_diffsView(diffsView),
+        m_apiPick(apiPick),
+        m_binaryOp(std::move(binaryOp)) {}
+
+  template <class MemberType>
+  KOKKOS_INLINE_FUNCTION void operator()(const MemberType& member) const {
+    const auto myRowIndex = member.league_rank();
+
+    auto myRowView  = Kokkos::subview(m_view, myRowIndex, Kokkos::ALL());
+    auto myDiffView = Kokkos::subview(m_diffsView, myRowIndex, Kokkos::ALL());
+
+    switch (m_apiPick) {
+      case 0: {
+        KE::adjacent_difference(member, KE::begin(myRowView),
+                                KE::end(myRowView), KE::begin(myDiffView));
+        break;
+      }
+
+      case 1: {
+        KE::adjacent_difference(member, KE::begin(myRowView),
+                                KE::end(myRowView), KE::begin(myDiffView),
+                                m_binaryOp);
+        break;
+      }
+
+      case 2: {
+        KE::adjacent_difference(member, myRowView, myDiffView);
+        break;
+      }
+
+      case 3: {
+        KE::adjacent_difference(member, myRowView, myDiffView, m_binaryOp);
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(std::size_t numTeams, std::size_t numCols, int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level adjacent_difference
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  auto [dataView, cloneOfDataViewBeforeOp_h] =
+      create_random_view_and_host_clone(
+          LayoutTag{}, numTeams, numCols,
+          Kokkos::pair{ValueType(5), ValueType(523)}, "dataView");
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  // to verify that things work, each team stores the result
+  // of its adjacent_difference call, and then we check
+  // that these match what we expect
+  Kokkos::View<ValueType**> diffsView("diffsView", numTeams, numCols);
```

Comment:
here diffsView is storing the result of the operation, but you also need a view with "distances" that you use to store the distance from begin of the iterator returned by the algorithm

Reply:
Done.